.github/workflows/ci.yml

```diff
@@ -712,6 +723,49 @@ jobs:
             gcr.io/iree-oss/emscripten@sha256:8ccc1c8de11919faf23aeaa585b13e5c5050952db76c101d6f61367280a3546f \
             build_tools/cmake/build_runtime_emscripten.sh
 
+  test_benchmark_suites:
+    needs: [setup, benchmarks]
```

Comment:
I'd rather we defer the changes to CI workflows to a second PR. Add the new test function and tests using it in one step, then run those tests on CI.

This is inverted - it doesn't make sense to run benchmarks _then_ test the correctness of those programs. Architecturally we should flip that around, and practically we should try to keep the multi-stage pipelines as flat as possible. The `benchmarks` job is already one of the last running jobs, so I'd prefer to not add yet another job that in turn depends on it: 
<img width="979" alt="image" src="https://user-images.githubusercontent.com/4010439/193158980-4b2357db-cc93-40c5-80e6-90aaab5ce00b.png">
(from https://github.com/iree-org/iree/actions/runs/3146189398)

Reply:
I understand your concern, and I also agree that the current flow is awkward. Ideally, we would much prefer we don't extend the critical CI path. However, we do want to reuse the artifacts built by one flow instead of building it again elsewhere (the current approach in `build_tools/cmake/test_riscv.sh`). Maybe the flow can be less awkward after the benchmark framework rework.

Given the need to have a safeguard mechanism in place for RISC-V as soon as possible, I would very much prefer to introduce the e2e `iree_run_module_test` check in CI sooner than later. We can revisit the flow again after the benchmark rework.

Reply:
Address the speedup suggestions in https://github.com/iree-org/iree/pull/10651